### PR TITLE
Add Deploy Scripts & S3Requester

### PR DESCRIPTION
**Pull request by justinkonecny**
- Add deployment scripts
- Add `S3Requester` and associated files
- Add `SLogger`
- Updates to dependencies

### CODE REVIEW

File: common/src/main/java/com/codeforcommunity/propertiesLoader/PropertiesLoader.java [hunk 1]
```diff
@@ -28,4 +44,16 @@ public static Properties getDbProperties() {
   public static Properties getExpirationProperties() {
     return getProperties("expiration.properties");
   }
+
```
Comment (jackblanc) [inline common/src/main/java/com/codeforcommunity/propertiesLoader/PropertiesLoader.java:47]: 🌱 💭  I still think I'd prefer one properties file

File: common/src/main/resources/properties/emailer.properties.example [hunk 2]
```diff
@@ -0,0 +1,9 @@
+mail.smtp.host = place-holder
+user = place-holder
+password = place-holder
+mail.smtp.auth = true
+mail.smtp.port = 465
+mail.debug = true
+mail.smtp.socketFactory.port = 465
+mail.smtp.socketFactory.class = javax.net.ssl.SSLSocketFactory
+mail.smtp.socketFactory.fallback = false
```
Comment (jackblanc) [inline common/src/main/resources/properties/emailer.properties.example:9]: ⛏️  newline

File: persist/pom.xml [hunk 3]
```diff
@@ -124,7 +124,7 @@
             <schema>public</schema>
           </schemas>
           <encoding>ISO-8859-1</encoding>
-          <cleanOnValidationError>true</cleanOnValidationError>
+          <cleanDisabled>true</cleanDisabled>
```
Comment (jackblanc) [inline persist/pom.xml:127]: ⛏️  This is a very important setting, can you add a comment here explaining this?

File: service/pom.xml [hunk 4]
```diff
@@ -94,6 +94,11 @@
...
       <scope>test</scope>
     </dependency>
+
+    <dependency>
+      <groupId>com.amazonaws</groupId>
+      <artifactId>aws-java-sdk-s3</artifactId>
+    </dependency>
   </dependencies>
 
 </project>
```
Comment (jackblanc) [inline service/pom.xml:104]: ⛏️ newline

File: service/src/main/java/com/codeforcommunity/requester/S3Requester.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+   * Events.
+   *
+   * @param eventTitle the title of the Event.
+   * @param base64Encoding the encoded image to upload.
+   * @return null if the initial base64Encoding was null, or the image URL if the upload was
+   *     successful.
+   * @throws BadRequestImageException if the base64 decoding failed.
+   * @throws S3FailedUploadException if the upload to S3 failed.
+   */
+  public static String validateUploadImageToS3LucyEvents(String eventTitle, String base64Encoding)
```
Comment (jackblanc) [inline service/src/main/java/com/codeforcommunity/requester/S3Requester.java:157]: ⛏️ LucyEvents

**Review by jackblanc — COMMENTED** [review]
This is awesome 🔥 
Left some questions

File: api/src/main/java/com/codeforcommunity/ApiMain.java [hunk 6]
```diff
@@ -1,26 +1,59 @@
...
+                .allowedHeader("Access-Control-Allow-Credentials")
+                .allowedHeader("Access-Control-Allow-Headers")
+                .allowedHeader("Access-Control-Request-Method")
+                .allowedHeader("X-Access-Token")
+                .allowedHeader("X-Refresh-Token"));
+
+    Route homeRoute = router.route("/");
+    homeRoute.handler(this::handleHealthCheck);
+
     router.mountSubRouter("/api/v1", apiRouter.initializeRouter(vertx));
```
[the remark this answers is not in the document]
Comment (justinkonecny) [reply api/src/main/java/com/codeforcommunity/ApiMain.java:50]: I honestly don't anticipate us ever having anything other than a v1

**Review by jackblanc — APPROVED** [review]
Let's merge this to unblock

**Pull request merged**